### ISSUE TITLE
The (ColorSchemeProvider) caused the re-render twice!

### ISSUE THREAD

**Issue by deounix** [mantinedev/mantine#814]
### What package has an issue

@mantine/styles

### Describe the bug

```
export function ColorSchemeProvider({
  colorScheme,
  toggleColorScheme,
  children,
}: ColorSchemeProviderProps) {
  const [key, setKey] = useState('init');

  useEffect(() => {
    setKey('update');
  }, []);

  return (
    <ColorSchemeContext.Provider key={key} value={{ colorScheme, toggleColorScheme }}>
      {children}
    </ColorSchemeContext.Provider>
  );
}
```

The problem here is that we're using the setKey for no reason! so it cause the re-render twice in every page. for example if we use:

```
useEffect(() => {
 console.log("HELLO"); // this will be called twice
}, []);
```

### In which browser did the problem occur

chrome

### Possible fix

```
export function ColorSchemeProvider({
  colorScheme,
  toggleColorScheme,
  children,
}: ColorSchemeProviderProps) {
  return (
    <ColorSchemeContext.Provider key="mantine-color-provider" value={{ colorScheme, toggleColorScheme }}>
      {children}
    </ColorSchemeContext.Provider>
  );
}
```

**Comment by rtivital** [comment]
It's rendered twice because some strategies that require checking color scheme on client will not work without rerender. For example, it is required on Mantine docs because we use use-local-storage-value hook, which does not trigger rerender by itself. This applies to every case when you can have mismatch of color scheme on client and server. If that is not what you need for your application, you can set up your own context logic which will not rerender.

**Comment by FabienDeborde** [comment]
I stumble on this issue after quite some time trying to debug my app...
I didn't paid much attention to the rerenders when I was rendering a few static lists, but once I started fetching data from external APIs I could definitely see that I was calling the API at least twice.

Like @rtivital said, it is up to the data storing / toggling strategy to prevent the rerenders.
I went with a "classic" redux approach, storing value in the store as well as in the localStorage, and initializing the store with localStorage value, and no more rerenders!

It would be really nicet hough, if the documentation had a small warning concerning the potential risks of rerenders when using context strategy.
(https://mantine.dev/theming/dark-theme/#colorschemeprovider)
